### ISSUE TITLE
Function parameter annotations

### ISSUE THREAD

**Issue by vadi2** [edbee/edbee-lib#69]
Saw in Android Studio for the first time - I think this is brilliant:

![selection_280](https://user-images.githubusercontent.com/110988/34205558-29cba126-e583-11e7-9239-7c0645f082df.png)

**Comment by gamecreature** [comment]
Do you know which c++ version is required for this?
Because I don't want to force users to use c++11

**Comment by vadi2** [comment]
I don't think this has anything to do with the C++ version, this is Java code. Just showing the IDE's ability to tell me what a random boolean parameter is!

**Comment by gamecreature** [comment]
Is this inserted in the editor temporary?

To be able to do this there's a lot of work to do.
We need to have a descent language parser on the background (langserver?)
Temporary inserting/showing the info about arguments

**Comment by keneanung** [comment]
More than understanding the syntax and language, this needs an understanding of documentation, as this is the source of the parameter name
